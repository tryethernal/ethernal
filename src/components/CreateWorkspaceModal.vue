<template>
<v-dialog v-model="dialog" max-width="700">
    <v-card>
        <v-card-title class="headline">
            New Workspace
            <v-spacer></v-spacer>
            <v-btn icon @click="close(false)"><v-icon>mdi-close</v-icon></v-btn>
        </v-card-title>
        <Create-Workspace @workspaceCreated="onWorkspaceCreated" @goToBilling="goToBilling" />
    </v-card>
</v-dialog>
</template>
<script>
import CreateWorkspace from './CreateWorkspace';

export default {
    name: 'CreateWorkspaceModal',
    components: {
        CreateWorkspace
    },
    data: () => ({
        dialog: false,
        resolve: null,
        reject: null,
    }),
    methods: {
        open: function() {
            this.dialog = true;
            return new Promise((resolve, reject) => {
                this.resolve = resolve;
                this.reject = reject;
            })
        },
        close: function(workspaceCreated = false) {
            const resolve = this.resolve;
            this.reset();
            resolve(workspaceCreated);
        },
        onWorkspaceCreated: function(workspaceData) {
            this.server.setCurrentWorkspace(workspaceData.name)
                .then(() => document.location = '/blocks');
        },
        goToBilling: function() {
            this.close(false);
            this.$router.push({ path: '/settings', query: { tab: 'billing' }});
        },
        reset: function() {
            this.dialog = false;
            this.resolve = null;
            this.reject = null;
        }
    }
}
</script>
