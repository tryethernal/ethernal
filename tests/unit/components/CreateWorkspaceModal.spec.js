import flushPromises from 'flush-promises';
import MockHelper from '../MockHelper';
  
import CreateWorkspaceModal from '@/components/CreateWorkspaceModal.vue';

describe('CreateWorkspaceModal.vue', () => {
    let helper;

    beforeEach(() => {
        helper = new MockHelper();
    });

    it('Should let the user create a new workspace', async () => {
        const wrapper = helper.mountFn(CreateWorkspaceModal, {
            stubs: ['Create-Workspace']
        });

        await wrapper.setData({ dialog: true, resolve: jest.fn() });

        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
