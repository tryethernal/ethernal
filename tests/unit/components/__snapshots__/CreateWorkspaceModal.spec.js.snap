// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWorkspaceModal.vue Should let the user create a new workspace 1`] = `
<div class="v-dialog__container">
  <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div tabindex="0" class="v-dialog v-dialog--active" style="transform-origin: center center; max-width: 700px;">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title headline">
            New Workspace
            <div class="spacer"></div> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--light"></i></span></button>
          </div>
          <create-workspace-stub></create-workspace-stub>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
